types/wordpress__dependency-extraction-webpack-plugin/index.d.ts

```diff
@@ -0,0 +1,18 @@
+// Type definitions for @wordpress/dependency-extraction-webpack-plugin 1.0
+// Project: https://github.com/WordPress/gutenberg/tree/master/packages/dependency-extraction-webpack-plugin/README.md
+// Definitions by: Derek Sifford <https://github.com/dsifford>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+// TypeScript Version: 3.4
+
+import { Plugin } from "webpack";
+
+export interface Options {
+    useDefaults?: boolean;
+    injectPolyfill?: boolean;
+    requestToExternal?(request: string): string | string[] | void;
+    requestToHandle?(request: string): string | void;
+}
+
+export default class DependencyExtractionWebpackPlugin extends Plugin {
```

Comment:
This package was declared using `module.exports = ...` or some variant. It should be declared with an `export = ...` like one of the following:

### Variables

```ts
declare var xyz: SomeType;

export = xyz;
```

### Classes

```ts
class Foo {
  // ...
}

export = Foo
```

### Functions

```ts
function foo(): any;

export = foo;
```

To import the entity using a default import, you will need to enable `esModuleInterop`. Alternatively, you can just import using a more appropriate syntax like

```ts
import foo = require("foo");
```

[See more here in the FAQ](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/9f9f6430117d025aef62ed56128deb65508edc86/README.md#a-package-uses-export--but-i-prefer-to-use-default-imports-can-i-change-export--to-export-default).

Reply:
Awesome explanation. Thanks for taking the time to write this out. I'll go ahead and fix. :+1: